hvplot/converter.py

```diff
@@ -2041,7 +2042,20 @@ def labels(self, x=None, y=None, data=None):
         self.use_index = False
         data, x, y = self._process_chart_args(data, x, y, single_y=True)
 
-        text = self.kwds.get('text', [c for c in data.columns if c not in (x, y)][0])
+        text = self.kwds.get('text')
+        if not text:
+            text = [c for c in data.columns if c not in (x, y)][0]
```

Comment:
Edge-case but if the first column that is not x/y contains a string with `'{foo}'` this might cause some problems?

Reply:
I'm not sure I follow; can you clarify this point?

Reply:
If I call `df.hvplot.labels()` without passing `text=...` it will default to the first column that is not x or y, if that column name is e.g. `'Volume {m3}` (edge-case-y, I know!), the code will break as it'll assume `m3` is a column name and find it missing.

Reply:
Okay I added support for this edge case: if there's only one missing col, and there's only one {} in the col, it uses the legacy behavior.

It does not support Volume {m3} {m4} though.

Reply:
I believe this could be approached differently by just skipping the whole formatting code path if `text` is not provided.